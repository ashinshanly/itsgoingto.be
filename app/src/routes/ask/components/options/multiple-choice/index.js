import MultipleChoice from './multiple-choice'

export default MultipleChoice
