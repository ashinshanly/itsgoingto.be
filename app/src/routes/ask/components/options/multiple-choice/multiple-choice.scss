@import "variables";

.input-option.input-option-multipleChoice {
  grid-column: 1 / span 1;

  .input-label-multipleChoice {
    line-height: 1em;
    font-size: 1.2em;
    padding-left: 0.7em;
    color: $base-font-color;
  }
}
