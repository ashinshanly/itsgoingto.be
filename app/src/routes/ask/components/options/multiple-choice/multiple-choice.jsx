import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { updatePoll } from 'store/poll'
import './multiple-choice.scss'

class MultipleChoice extends PureComponent {
  handleMultipleChoiceChange = (event) =>
    this.props.updateOptions({
      identifier     : '',
      multipleChoice : event.target.checked
    })

  render () {
    const { poll } = this.props

    return (
      <div className='input-option input-option-multipleChoice'>
        <input
          id='multiple-choice'
          className='input-checkbox input-checkbox-multipleChoice'
          name='multiple'
          type='checkbox'
          checked={poll.multipleChoice}
          onChange={this.handleMultipleChoiceChange} />
        <label
          htmlFor='multiple-choice'
          className='input-label input-label-options input-label-multipleChoice'>
          Multiple choice responses
        </label>
      </div>
    )
  }
}

MultipleChoice.propTypes = {
  poll          : PropTypes.object.isRequired,
  updateOptions : PropTypes.func.isRequired
}

const mapDispatchToProps = (dispatch) => ({
  updateOptions: (value) => dispatch(updatePoll(value))
})

export default connect(null, mapDispatchToProps)(MultipleChoice)
