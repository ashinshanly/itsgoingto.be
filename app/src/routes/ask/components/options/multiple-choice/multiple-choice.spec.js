/* global expect, jest */
import React from 'react'
import { shallow } from 'enzyme'
import { MultipleChoice } from './multiple-choice'

const props = {
  poll: { multipleChoice: false },
  updateOptions: jest.fn(),
}

describe('(Route) Ask', () => {
  describe('(Component) Multiple Choice', () => {
    describe('(Action) onChange', () => {
      it('should call prop onAnswerChange', () => {
        const wrapper = shallow(<MultipleChoice {...props} />)
        const event = {
          preventDefault() { },
          target: { checked: true }
        }

        wrapper.find('input').simulate('change', event)
        expect(props.updateOptions).toHaveBeenCalledWith({
          identifier: '',
          multipleChoice: true,
        })
      })
    })

    describe('(Render)', () => {
      it('matches snapshot', () => {
        const wrapper = shallow(<MultipleChoice {...props} />)

        expect(wrapper).toMatchSnapshot()
      })
    })
  })
})
