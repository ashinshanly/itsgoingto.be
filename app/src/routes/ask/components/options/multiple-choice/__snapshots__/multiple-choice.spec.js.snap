// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Route) Ask (Component) Multiple Choice (Render) matches snapshot 1`] = `
<div
  className="input-option input-option-multipleChoice"
>
  <input
    checked={false}
    className="input-checkbox input-checkbox-multipleChoice"
    id="multiple-choice"
    name="multiple"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="input-label input-label-options input-label-multipleChoice"
    htmlFor="multiple-choice"
  >
    Multiple choice responses
  </label>
</div>
`;
